### PR TITLE
Watch for changes to html.js

### PR DESCRIPTION
**Pull request by felixjung**
Addresses #1442.

@KyleAMathews, I think we can address the configuration via socket support in a separate topic branch?

### CODE REVIEW

**Comment by KyleAMathews** [comment]
>Won't StaticSiteGeneratorPlugin automatically rebuild all pages?

No check out develop-html.js — it only builds one index.html page.

>The only thing I see right now is to include socket.io and push an event to the client whenever index.html has been rebuilt. The client can then refresh the page.

Yeah exactly. Gatsby's entry for development would connect to the dev server and then would refresh when instructed.

**Comment by KyleAMathews** [comment]
Looking good so far!

**Comment by KyleAMathews** [comment]
Deploy preview failed.

Built with commit d607b3e81176053c02993fc66685579218834d0e

https://app.netlify.com/sites/using-styled-components/deploys/5965d5216f4c507bc0303655

**Comment by KyleAMathews** [comment]
Deploy preview failed.

Built with commit d607b3e81176053c02993fc66685579218834d0e

https://app.netlify.com/sites/using-glamor/deploys/5965d5216f4c507bc0303652

**Comment by KyleAMathews** [comment]
Deploy preview failed.

Built with commit d607b3e81176053c02993fc66685579218834d0e

https://app.netlify.com/sites/using-postcss-sass/deploys/5965d5206f4c507bc0303649

**Comment by KyleAMathews** [comment]
Deploy preview failed.

Built with commit d607b3e81176053c02993fc66685579218834d0e

https://app.netlify.com/sites/image-processing/deploys/5965d5216f4c507bc030364f

**Comment by KyleAMathews** [comment]
Deploy preview failed.

Built with commit d607b3e81176053c02993fc66685579218834d0e

https://app.netlify.com/sites/using-contentful/deploys/5965d5216f4c507bc0303658

**Comment by felixjung** [comment]
OK. This works end-to-end now. @KyleAMathews, I'm not sure if I put the client code in the right places. Could you review the code please? ☺️

**Comment by felixjung** [comment]
Hah, I guess I still have to add some tests :wink:

File: packages/gatsby/src/cache-dir/app.js [hunk 1]
```diff
@@ -8,6 +8,13 @@ window.___emitter = require(`./emitter`)
 // Let the site/plugins run code very early.
 apiRunner(`onClientEntry`)
 
+if (process.env.NODE_ENV === `development`) {
```
Comment (KyleAMathews) [inline packages/gatsby/src/cache-dir/app.js]: This isn't necessary as this file is only used in development.

Comment (felixjung) [inline packages/gatsby/src/cache-dir/app.js]: Ah, good to know. I wasn't sure, but I guess I should have taken a closer look at the webpack config 😁

**Comment by felixjung** [comment]
I've added tests for the `src/util/path.js` module. I have 0️⃣ clue how to test the develop server though 😞 That being said, I don't see too many tests in there anyway 😂

**Comment by felixjung** [comment]
Wow, getting those windows tests right took longer than it should have 🙄

**Comment by KyleAMathews** [comment]
Just tested it, worked perfectly! Slick work!

The only change I'd suggest making is to add a console.log when Gatsby starts rebuilding the html.js file so the user knows something is happening and why.

**Comment by KyleAMathews** [comment]
I'm going to do another minor release now and want this in it so I'll just merge this but please do the above in a follow-up PR.

Thanks for taking this on! Hot reload all the things :-)

**Comment by felixjung** [comment]
Woohoo! Yeah, what you say makes a lot of sense 😀.

**Comment by jlengstorf** [comment]
Hiya @felixjung! 👋

This is definitely late, but on behalf of the entire Gatsby community, I wanted to say thank you for being here.

Gatsby is built by awesome people like you. Let us say “thanks” in two ways:

1.  **We’d like to send you some Gatsby swag.** As a token of our appreciation, you can go to the [Gatsby Swag Store][store] and log in with your GitHub account to get a coupon code good for one free piece of swag. (We’ve got t-shirts and hats, plus some socks that are really razzing our berries right now.)
2.  **If you’re not already part of it, we just invited you to join the Gatsby organization on GitHub.** This will add you to our team of maintainers. You’ll receive an email shortly asking you to confirm. By joining the team, you’ll be able to label issues, review pull requests, and merge approved pull requests.

If you have questions, please don’t hesitate to reach out to us: tweet at [@gatsbyjs][twitter] and we’ll come a-runnin’.

Thanks again! 💪💜

[store]: https://store.gatsbyjs.org
[twitter]: https://twitter.com/gatsbyjs

**Comment by felixjung** [comment]
Thanks a ton @jlengstorf! I had heard about this in various podcasts, but wasn’t expecting it anymore. What a nice surprise. Keep it up!

**Pull request merged**